things/model/src/main/java/org/eclipse/ditto/things/model/signals/events/ThingMigrated.java

```diff
@@ -0,0 +1,222 @@
+     * @param value            The value describing the changes that were migrated.
+     * @param revision         The revision number of the thing.
+     * @param timestamp        The event timestamp.
+     * @param dittoHeaders     The Ditto headers.
+     * @param metadata         The metadata associated with the event.
+     * @return The created {@code ThingMigrated} event.
+     */
+    public static ThingMigrated of(final ThingId thingId,
+            final JsonPointer path,
+            final JsonValue value,
+            final long revision,
+            @Nullable final Instant timestamp,
+            final DittoHeaders dittoHeaders,
+            @Nullable final Metadata metadata) {
+        return new ThingMigrated(thingId, path, value, revision, timestamp, dittoHeaders, metadata);
+    }
+
+    /**
+     * Creates a new {@code ThingMigrated} event from a JSON object.
+     *
+     * @param jsonObject   The JSON object from which the event is created.
+     * @param dittoHeaders The headers of the command.
+     * @return The {@code ThingMigrated} event created from JSON.
+     */
+    public static ThingMigrated fromJson(final JsonObject jsonObject, final DittoHeaders dittoHeaders) {
+        return new EventJsonDeserializer<ThingMigrated>(TYPE, jsonObject).deserialize(
+                (revision, timestamp, metadata) -> {
+                    final ThingId thingId = ThingId.of(jsonObject.getValueOrThrow(ThingEvent.JsonFields.THING_ID));
+                    final JsonPointer path = JsonPointer.of(jsonObject.getValueOrThrow(JsonFields.JSON_PATH));
+                    final JsonValue value = jsonObject.getValueOrThrow(JsonFields.JSON_VALUE);
+
+                    return of(thingId, path, value, revision, timestamp, dittoHeaders, metadata);
+                });
+    }
+
+    @Override
+    protected void appendPayload(final JsonObjectBuilder jsonObjectBuilder,
+            final JsonSchemaVersion schemaVersion, final Predicate<JsonField> predicate) {
+        jsonObjectBuilder.set(ThingEvent.JsonFields.THING_ID, thingId.toString(), predicate);
+        jsonObjectBuilder.set(JsonFields.JSON_PATH, path.toString(), predicate);
+        jsonObjectBuilder.set(JsonFields.JSON_VALUE, value, predicate);
+    }
+
+    @Override
+    public ThingMigrated setDittoHeaders(final DittoHeaders dittoHeaders) {
+        return of(thingId, path, value, getRevision(), getTimestamp().orElse(null), dittoHeaders,
+                getMetadata().orElse(null));
+    }
+
+    @Override
+    public Command.Category getCommandCategory() {
+        return Command.Category.MODIFY;
+    }
+
+    @Override
+    public ThingMigrated setRevision(final long revision) {
+        return of(thingId, path, value, revision, getTimestamp().orElse(null), getDittoHeaders(),
+                getMetadata().orElse(null));
+    }
+
+    @Override
+    public JsonPointer getResourcePath() {
+        return path;
+    }
+
+    public JsonValue getValue() {
+        return value;
```

Comment:
What does this `value` represent?
This is too technical for me - I don't understand only looking at this class what "value" does or should persist.

Part of the problem is that public getters are not documented ;)